projects/rero/ng-core/src/lib/record/editor/editor.component.ts

```diff
@@ -118,13 +117,43 @@ export class EditorComponent implements OnInit, OnChanges, OnDestroy {
     'dateMustBeLessThan'
   ];
 
+  // list of fields to be hidden
+  private _hiddenFields: FormlyFieldConfig[] = [];
+
+  // Observable of hidden fields
+  private _hiddenFieldsSubject: BehaviorSubject<FormlyFieldConfig[]> = new BehaviorSubject([]);
+
+  // current list of hidden fields
+  public get hiddenFields$(): Observable<any[]> {
+    return this._hiddenFieldsSubject.asObservable();
+  }
+
+  // Editor long mode
+  public get longMode(): boolean {
+    return this.editorSettings.longMode;
+  }
+
+  // Editor edit mode
+  public get editMode(): boolean {
+    return this.pid ? true : false;
+  }
+
+  // Editor root field
+  public get rootField(): FormlyFieldConfig {
+    return this.rootFormlyConfig;
+  }
+
+  // Editor fonction
+  public get editorComponent(): () => EditorComponent {
+    return () => this;
+  }
```

Comment:
for my information : I don't really understand the return type of this function :(

Reply:
Anonymous function that returns an instance of the EditorComponent